src/system.rs

```diff
@@ -31,3 +127,88 @@ mod unit_tests {
         assert_eq!(uptime_internal("12489513.08 22906637.29\n"), 12489513);
     }
 }
+
+/// Returns information about virtual memory usage
+///
+/// `/proc/meminfo` contains the virtual memory statistics
+pub fn virtual_memory() -> Option<VirtualMemory> {
+    let data = read_file(Path::new("/proc/meminfo")).unwrap();
```

Comment:
If this function returns Result, I think you can use `?` here - which makes it easier for the caller to handle cases where /proc/meminfo isn't found (whereas `unwrap` just panics on failure).